doc/migration.rst

```diff
@@ -114,3 +70,88 @@ it is not preserved for `dune-workspace` files. I.e. a
 `dune-workspace.in` file means nothing.
 
 This rule was only useful when we didn't have project files.
+
+Variable Syntax
+---------------
+
+``${foo} and $(foo)`` are no longer valid variable syntax in dune files.
+Variables are defined as ``%{foo}``. This change is done to simplify
+interoperability with bash commands which also use teh ``${foo}`` syntax.
+
+``(files_recursively_in ..)`` is removed
+----------------------------------------
+
+The ``files_recursively_in`` stanza is invalid in dune files. A
+:ref:`source_tree <source_tree>` stanza has been introduced to reflect the
+actual function of this stanza.
+
+Escape Sequences
+----------------
+
+Invalid escape sequences such as ``\999`` are not tolerated in quoted strings in
```

Comment:
There were not accepted in jbuild files either, only in block comments. One thing that we reject now and didn't before is `\ `, i.e. `\` followed by a character other that 0-9, x, n, r, t, b.